test/shared_test/lib_battery_voltage_test.cpp

```diff
@@ -291,34 +295,155 @@ TEST_F(voltage_dynamic_lib_battery_voltage_test, calculateMaxDischargeSubMinute)
         cap->updateCapacity(I, dt_hour);
     power = model->calculate_max_discharge_w(cap->q0(), cap->qmax(), 0, &max_current);
-    EXPECT_NEAR(power, 26689, 1);
+    EXPECT_NEAR(power, 14843, 1);
     max_current_calc = model->calculate_current_for_target_w(power, cap->q0(), cap->qmax(), 0);
     EXPECT_NEAR(max_current_calc, max_current, 0.6);
     // Does not empty battery for highest power
     cap->updateCapacity(max_current, dt_hour);
-    EXPECT_NEAR(cap->SOC(), 90.345, 1e-3);
+    EXPECT_NEAR(cap->SOC(), 92.736, 1e-3);
+}
+
+TEST_F(voltage_dynamic_lib_battery_voltage_cutoff_test, calculateMaxDischargeHourly) {
+    double dt_hour = 1;
+    CreateModel(dt_hour);
+
+    // start at half SOC
+    double max_current;
+    double power = model2->calculate_max_discharge_w(cap2->q0(), cap2->qmax(), 0, &max_current);
+    EXPECT_NEAR(power, 2373, 1);        // current ~4
+    //EXPECT_NEAR(power / max_current, 0, 1); //voltage test
+    double max_current_calc = model2->calculate_current_for_target_w(power, cap2->q0(), cap2->qmax(), 0);
+    EXPECT_NEAR(max_current_calc, max_current, 0.6);
+    //EXPECT_NEAR(power / max_current_calc, 0, 1); //voltage test 2
+    // Does not empty battery for highest power
+    cap2->updateCapacity(max_current, dt_hour);
+    EXPECT_NEAR(cap2->SOC(), 5, 1e-3);
+
+    // start at empty SOC
+    power = model2->calculate_max_discharge_w(cap2->q0(), cap2->qmax(), 0, &max_current);
+    EXPECT_NEAR(power, 217, 1);
+    //EXPECT_NEAR(power / max_current, 0, 1); //voltage test
+    max_current_calc = model2->calculate_current_for_target_w(power, cap2->q0(), cap2->qmax(), 0);
+    EXPECT_NEAR(max_current_calc, max_current, 0.3);
+    //EXPECT_NEAR(power / max_current_calc, 0, 1); //voltage test 2
+    // Empties battery for highest power
+    cap2->updateCapacity(max_current, dt_hour);
+    EXPECT_NEAR(cap2->SOC(), 5, 1e-3);
+
+    // start at full SOC
+    double I = -2;
+    while (cap2->SOC() < 95)
+        cap2->updateCapacity(I, dt_hour);
+    power = model2->calculate_max_discharge_w(cap->q0(), cap2->qmax(), 0, &max_current);
+    EXPECT_NEAR(power, 2373, 1);
+    max_current_calc = model2->calculate_current_for_target_w(power, cap2->q0(), cap2->qmax(), 0);
+    EXPECT_NEAR(max_current_calc, max_current, 0.7);
+    // Does not empty battery for highest power
+    cap2->updateCapacity(max_current, dt_hour);
+    EXPECT_NEAR(cap2->SOC(), 50, 1e-3);
+}
+
+TEST_F(voltage_dynamic_lib_battery_voltage_cutoff_test, calculateMaxDischargeSubHourly) {
+    double dt_hour = 0.5;
+    CreateModel(dt_hour);
+
+    // start at half SOC
+    double max_current;
+    double power = model2->calculate_max_discharge_w(cap2->q0(), cap2->qmax(), 0, &max_current);
+    EXPECT_NEAR(power, 4669, 1);        // current ~8
+    double max_current_calc = model2->calculate_current_for_target_w(power, cap2->q0(), cap2->qmax(), 0);
+    EXPECT_NEAR(max_current_calc, max_current, 2);
+    // Does not empty battery for highest power
```

Comment:
@brtietz @dguittet Running into some issues with calculate_current_for_target_w when creating new tests with non-zero voltage cutoffs. The solver this function uses is not aware of the change to Qfull, and my attempts to modify it have not improved the accuracy as of yet between the current results (notice the 2 amp difference between max_current_calc and max_current here).

Reply:
I think this function should be aware of Q_full_mod and/or V_cut. Is it possible to store them in voltage_params and pull them out again inside `calculate_current_for_target_w`?

Happy to meet to discuss, send me an invite if you think that's helpful (please avoid Wednesday if possible).